libraries/AP_Generator/AP_Generator_IE_FuelCell.h

```diff
@@ -26,6 +26,9 @@ class AP_Generator_IE_FuelCell : public AP_Generator_Backend
     // Update fuel cell, expected to be called at 20hz
     void update(void) override;
 
+    // Allow generator error code to pass up to frontend and export to LUA
+    uint32_t get_errorcode(void) override {return _err_code;}
```

Comment:
I'm *really* not a fan of returning driver-specifc error codes in a `uint32_t`.

Options include
 - creating an enumeration in `AP_Generator` which describes all possible bits which could be set, and which the backends can supply
 - create singletons for the different generator backends individually and add bindings for those singletones.